pandas/core/arrays/datetimelike.py

```diff
@@ -207,6 +208,24 @@ def astype(self, dtype, copy=True):
             return self._box_values(self.asi8)
         return super(DatetimeLikeArrayMixin, self).astype(dtype, copy)
 
+    def repeat(self, repeats, *args, **kwargs):
```

Comment:
Is a `repeat` method needed? For now, we don't have that in our EA interface, so if avoidable, I would keep it that way (and keep it on the index).

Reply:
`repeat` is also moved to PeriodArray in #22862.

Reply:
So we might need to discuss it there as well.

But also, in that PR, the index method still exists. So in this PR, you are removing something from the index, which will need to be added back in a next PR.